Assets/Tests/EditMode/AuthTests/RestRequestsAdapters/MetamaskRestRequestsAdapterTests.cs

```diff
@@ -0,0 +1,99 @@
+using System.Collections.Generic;
+using System.Net.Http;
+using System.Threading.Tasks;
+using Moq;
+using NUnit.Framework;
+using Src.AuthController.CredentialProviders.Metamask.MetamaskRestRequestsAdapter;
+using Src.AuthController.REST;
+using Src.AuthController.REST.REST_Request_Proxies.Metamask;
+using Src.AuthController.REST.REST_Response_DTOs.MetamaskBackend;
+
+namespace Tests.EditMode.AuthTests.RestRequestsAdapters
+{
+    public class MetamaskRestRequestsAdapterTests
+    {
+        private static string[] testStrings =
+        {
+            "asldkjfahsd",
+            "eoirt34w5r",
+            "2346539485y"
+        };
+        
+        [Test]
+        public async Task GetNonce_ShouldObtainNonce([ValueSource(nameof(testStrings))] string address)
+        {
+            var expectedNonceResponse = new MetamaskNonceResponse();
+            var nonceRequestParamsStub = new MetamaskNonceRequestDtoProxy(address);
+            
+            var httpClientRequestAdapterMock = new Mock<IHttpClientRequestAdapter>();
+            var tcs = new TaskCompletionSource<MetamaskNonceResponse>();
+
+            httpClientRequestAdapterMock.Setup(a => a.Request(
+                HttpMethod.Get,
+                It.IsAny<string>(),
```

Comment:
As far as I understand, the idea of MetamaskRestRequestsAdapter is that it calles HttePrequestAdapter with correct uri, so I think this fact should be tested here.